compiler/hash-typecheck/src/error.rs

```diff
@@ -336,13 +396,51 @@ impl TypecheckError {
                         "Unknown field",
                     )))
                     .add_element(ReportElement::Note(ReportNote::new(
-                        "note",
+                        ReportNoteKind::Note,
                         format!(
                             "The field `{}` doesn't exist on struct `{}`.",
                             name, ty_name
                         ),
                     )));
             }
+            TypecheckError::EnumVariantArgumentsMismatch {
+                ty_def_location,
+                ty_def_name,
+                variant_name,
+                location,
+                mismatch,
+            } => {
+                builder
+                    .with_error_code(HashErrorCode::UnresolvedStructField)
+                    .with_message("Enum variant arguments mismatch");
+
+                let name = IDENTIFIER_MAP.get_ident(variant_name);
+                let ty_name = IDENTIFIER_MAP.get_ident(ty_def_name);
+
+                // If we have the location of the definition, we can print it here
+                if let Some(ty_def_location) = ty_def_location {
+                    builder.add_element(ReportElement::CodeBlock(ReportCodeBlock::new(
+                        ty_def_location,
+                        format!("The enum variant `{}` is defined here.", ty_name),
+                    )));
+                }
+
+                builder
+                    .add_element(ReportElement::CodeBlock(ReportCodeBlock::new(
+                        location, "here",
+                    )))
+                    .add_element(ReportElement::Note(ReportNote::new(
+                        ReportNoteKind::Note,
+                        format!(
+                            "This variant `{}` has an invalid number of arguments.",
```

Comment:
Maybe s/invalid/incorrect/ ?
Also it might be useful to print the actual numbers.